indico/modules/events/registration/notifications.py

```diff
@@ -63,6 +64,8 @@ def notify_registration_modification(registration, notify_user=True):
 
 
 def notify_registration_state_update(registration):
-    _notify_registration(registration, 'registration_state_update_to_registrant.html')
+    attach_rejection_reason = request.form.get('attach_rejection_reason') == 'y'
```

Comment:
This should be passed as an argument as weld - the data should be coming from `form.attach_rejection_reason.data`, and not from manually reading the form data again..